server/routes/visit.ts

```diff
@@ -99,9 +99,13 @@ export default function routes({
       operationId: res.locals.appInsightsOperationId,
     })
 
+    const oneDayInMs = 1000 * 60 * 60 * 24
```

Comment:
Minor point but as you're using this as a fixed value, you could put it at the top of the file somewhere instead of inside a route so it's set once rather than being re-calculated for each request. (e.g. similar thing done in `SupportedPrisonService.ts`. 

Similarly, could also make it a bit more explicit/configurable where the '28' days is from. e.g. at the top something like `const CANCELLATION_LIMIT_DAYS=28` . 

The `date-fns` library has various conversion helpers and methods for adding / subtracting / comparing difference days. Don't know if you tried any of these; could well be overkill though I guess...